docs/sidechain/spec.md

```diff
@@ -0,0 +1,945 @@
+no exist. If the `SidechainAccountCreateState` does not already exist it will be created.
+
+#### Fields
+
+* Attestations. A collection of attestations, witnessing a particular event on
+  the other chain. This collection may be contain attestations for different
+  sidechain ledger objects. See section above for fields in an attestation.
+
+### SidechainModify
+
+Change the `SignaturesReward` field or the `MinAccountCreateAmount` on the
+sidechain object. Note that this is a regular transaction that is sent by the
+door account and requires the entities that control the witness servers to
+co-operate and provide the signatures for this transaction. This happens outside
+the ledger.
+
+The `SignaturesReward` and `MinAccountCreateAmount` of a transaction are the
+values that were in effect at the time the transaction was submitted.
+
+Note that the signer's list for the sidechain is not modified through this
+transaction. The signer's list is on the door account itself and is changed in
+the same way signer's lists are changed on accounts.
+
+#### Fields
+
+The transaction contains the following fields:
+
+* Sidechain: Door accounts and assets. See `STSidechain` above. Required.
+
+* SignaturesReward: Total amount, in XRP, to be rewarded for providing a
+  signatures for a cross-chain transfer or for signing for the cross-chain
+  reward. This will be split among the signers. Optional.
+
+* MinAccountCreateAmount: Minimum Amount, in XRP, required for an
+  `SidechainXChainAccountCreate` transaction. If this is zero, the
+  field will be removed from the ledger object. Optional
+  
+At least one of `SignaturesReward` and `MinAccountCreateAmount` must be present.
+
+```c++
+    add(jss::SidechainModify,
+        ttSIDECHAIN_MODIFY,
+        {
+            {sfSidechain,              soeREQUIRED},
+            {sfSignaturesReward,       soeOPTIONAL},
+            {sfMinAccountCreateAmount, soeOPTIONAL},
+        },
+        commonFields);
+```
+
+
+## New RPC Commands
+
+### Sidechain transaction history
+
+Subscribe to transactions that effect a sidechain object, similar to subscribing
+to transaction history. This would return any transaction that changes the state
+of the sidechain ledger object, including:
+
+* New sequence number checked-out
+* Funds put into trust
+* Funds issued
+* Reward collected
+* Min create fee changed
+
+Since the `SidechainAddWitness` command can batch signatures, batching all the
+signatures from a single ledger is a natural batch size. To do this, the witness
+server would need to know when it has collected all the transactions from a
+ledger. It would be good to add a field that either tells us how many of these
+transactions are in this ledger or a field that tells us when the transaction we
+receive is the last to be sent for this ledger.
+
+### Other RPC Commands 
+
+* Given a transaction id, get the sequence number
+* Given a sidechain description, get the sidechain ledger object
+
```

Comment:
A note here should be added about adding the extra filter to account_objects for the cross-chain sequence